### PR TITLE
Check for teleport launchd before installing through join script

### LINKED ISSUE #31554: Download script for server says it's not running (but it is)
Log says it's not "running", but it is and I can ssh into it (and testing connection in discover flow succeeds as well)

```
$ sudo bash -c "$(curl -fsSL https://kimlisa.cloud.gravitational.io/scripts/f798eac77700bb26387272abd0528b9c/install-node.sh)"
Password:
2023-09-06 19:38:00 PDT [teleport-installer] TELEPORT_VERSION: 14.0.0-alpha.2
2023-09-06 19:38:00 PDT [teleport-installer] TARGET_HOSTNAME: kimlisa.cloud.gravitational.io
2023-09-06 19:38:00 PDT [teleport-installer] TARGET_PORT: 443
2023-09-06 19:38:00 PDT [teleport-installer] JOIN_TOKEN: f798eac77700bb26387272abd0528b9c
2023-09-06 19:38:00 PDT [teleport-installer] CA_PIN_HASHES: sha256:bca5a029792d211e734df45d45ec1ace4b39dc7837664fc7cb313953ac2c3fe2
2023-09-06 19:38:00 PDT [teleport-installer] Checking TCP connectivity to Teleport server (kimlisa.cloud.gravitational.io:443)
2023-09-06 19:38:00 PDT [teleport-installer] Connectivity to Teleport server (via nc) looks good
2023-09-06 19:38:00 PDT [teleport-installer] Detected host: darwin22.1.0, using Teleport binary type darwin
2023-09-06 19:38:00 PDT [teleport-installer] Detected MacOS arm64 architecture, using Teleport arch arm64
2023-09-06 19:38:00 PDT [teleport-installer] Using Teleport distribution: tarball
2023-09-06 19:38:00 PDT [teleport-installer] Created temp dir /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/teleport-XXXXXXXXXX.WXHmndpy
2023-09-06 19:38:00 PDT [teleport-installer] Installing from binary file.
2023-09-06 19:38:00 PDT [teleport-installer] Downloading Teleport tarball release 14.0.0-alpha.2
2023-09-06 19:38:00 PDT [teleport-installer] Running curl -fsSL --retry 5 --retry-delay 5 https://get.gravitational.com/teleport-ent-v14.0.0-alpha.2-darwin-arm64-bin.tar.gz
2023-09-06 19:38:00 PDT [teleport-installer] Downloading to /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/teleport-XXXXXXXXXX.WXHmndpy/teleport-ent-v14.0.0-alpha.2-darwin-arm64-bin.tar.gz
2023-09-06 19:38:08 PDT [teleport-installer] Downloaded file size: 148890658 bytes
2023-09-06 19:38:08 PDT [teleport-installer] Will use shasum -a 256 to validate the checksum of the downloaded file
2023-09-06 19:38:09 PDT [teleport-installer] The downloaded file's checksum validated correctly
2023-09-06 19:38:12 PDT [teleport-installer] Found: Teleport Enterprise v14.0.0-alpha.2 git:v14.0.0-alpha.2-0-g71aea8d go1.21.0
2023-09-06 19:38:12 PDT [teleport-installer] Writing Teleport node service config to /etc/teleport.yaml

A Teleport configuration file has been created at "/etc/teleport.yaml".
Add your Teleport license file to "/var/lib/teleport/license.pem".
To start Teleport with this configuration file, run:

sudo teleport start --config="/etc/teleport.yaml"

Note that starting a Teleport server with this configuration will require root access as:
- The Teleport configuration is located at "/etc/teleport.yaml".
- Teleport will be storing data at "/var/lib/teleport". To change that, run "teleport configure" with the "--data-dir" flag.

Happy Teleporting!
2023-09-06 19:38:12 PDT [teleport-installer] Host is running MacOS
2023-09-06 19:38:12 PDT [teleport-installer] Installing Teleport launchd config to /Library/LaunchDaemons
2023-09-06 19:38:12 PDT [teleport-installer] Starting Teleport via launchctl. It will automatically be started whenever the system reboots.
/Library/LaunchDaemons/com.goteleport.teleport.plist: service already loaded
Load failed: 37: Operation already in progress

2023-09-06 19:38:15 PDT [teleport-installer] ---> The Teleport service was installed, but it does not appear to have started successfully.
2023-09-06 19:38:15 PDT [teleport-installer] ---> Check Teleport logs in '/var/log/teleport-stderr.log' and '/var/log/teleport-stdout.log'
2023-09-06 19:38:15 PDT [teleport-installer] ---> Contact Teleport support for further assistance.
```

Current behavior:

Bug details:
- Teleport version: Teleport Enterprise v14.0.0-alpha.2 git:v14.0.0-alpha.2-0-g71aea8d go1.21.0
- Recreation steps: use the discover flow (pictured below) to add a server (I used mac m1)
image

### PR DESCRIPTION
**Pull request by kimlisa**
resolves https://github.com/gravitational/teleport/issues/31554

Checks for teleport launchd (mac) before attempting to install teleport. Otherwise user's get a false "failed to start" message. This is only an issue if teleport was previously installed

#### Mac screenshot

mac when launchd plist file exists:

image

after success:

image

#### Ubuntu screenshot (to make sure changes to mac related is not affecting linux related)

when teleport already exists:

image

after success:
image

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]

File: lib/web/scripts/node-join/install.sh [hunk 1]
```diff
@@ -572,6 +578,8 @@ teleport_binaries_exist() {
 teleport_config_exists() { if [ -f ${TELEPORT_CONFIG_PATH} ]; then return 0; else return 1; fi; }
 # checks whether a teleport data dir exists on the host
 teleport_datadir_exists() { if [ -d ${TELEPORT_DATA_DIR} ]; then return 0; else return 1; fi; }
+# checks whether a teleport launch daemon plist files exists on the host
```
Comment (webvictim) [inline lib/web/scripts/node-join/install.sh:581]:
```suggestion
# checks whether a launchd plist file for teleport already exists on the host
```

File: lib/web/scripts/node-join/install.sh [hunk 2]
```diff
@@ -690,6 +698,11 @@ elif [[ "${OSTYPE}" == "darwin"* ]]; then
     fi
     log "Detected macOS ${ARCH} architecture, using Teleport arch ${TELEPORT_ARCH}"
     TELEPORT_FORMAT="tarball"
+    if launchd_plist_file_exists; then
+        log_header "Warning: Found existing Teleport launchd ${LAUNCHD_CONFIG_PATH}/${LAUNCHD_PLIST_FILE}."
```
Comment (webvictim) [inline lib/web/scripts/node-join/install.sh:702]:
```suggestion
        log_header "Warning: Found existing Teleport launchd config ${LAUNCHD_CONFIG_PATH}/${LAUNCHD_PLIST_FILE}."
```

File: lib/web/scripts/node-join/install.sh [hunk 3]
```diff
@@ -237,6 +238,11 @@ log_cleanup_message() {
     log_only "- remove any Teleport binaries (${TELEPORT_BINARY_LIST}) installed under ${TELEPORT_BINARY_DIR}"
     for BINARY in ${TELEPORT_BINARY_LIST}; do EXAMPLE_DELETE_COMMAND+="${TELEPORT_BINARY_DIR}/${BINARY} "; done
     log_only "  - rm -f ${EXAMPLE_DELETE_COMMAND}"
+    if is_macos_host; then
+        log_only "- unload and remove Teleport launchd ${LAUNCHD_CONFIG_PATH}/${LAUNCHD_PLIST_FILE}"
```
Comment (webvictim) [inline lib/web/scripts/node-join/install.sh:242]:
```suggestion
        log_only "- unload and remove Teleport launchd config ${LAUNCHD_CONFIG_PATH}/${LAUNCHD_PLIST_FILE}"
```

**Review by webvictim — APPROVED** [review]
Few tiny nits!

**Review by avatus — APPROVED** [review]

**Pull request merged**